reason-react-navigation/src/ReactNavigation.re

```diff
@@ -319,6 +1140,596 @@ module Make = (S: {type screenProps;}) => {
     "navigationOptions";
   [@bs.set]
   external setDynamicNavigationOptions:
-    (React.component('props), navigationParams => NavigationOptions.t) => unit =
+    (
+      React.component('props),
+      NavigationScreenProps.t('params, 'options) => NavigationOptions.t
+    ) =>
+    unit =
     "navigationOptions";
-};
\ No newline at end of file
+
+  module Route = (M: Route) => {
+    include NavigationBase(M);
+    let navigate = navigation => navigate(navigation, M.routeName);
+    [@bs.send]
+    external navigateWithParams: (t, string, M.params) => unit = "navigate";
+    let navigateWithParams = (navigation, params: M.params) =>
+      navigateWithParams(navigation, M.routeName, params);
+
+    let getParams = (navigation: t, ()): M.params =>
+      navigation->state##params;
+
+    [@bs.set]
+    external setNavigationOptions:
+      (React.component('props), NavigationOptions.t) => unit =
+      "navigationOptions";
+    [@bs.set]
+    external setDynamicNavigationOptions:
+      (
+        React.component('props),
+        NavigationScreenProps.t(M.params, 'options) => NavigationOptions.t
+      ) =>
+      unit =
+      "navigationOptions";
+  };
+
+  [@bs.val] [@bs.module "@react-navigation/core"]
+  external navigationContext: React.Context.t(navigation) =
+    "NavigationContext";
+
+  module type Config = {
+    type context;
+    let defaultValue: context;
+  };
+
+  let useNavigation = () => React.useContext(navigationContext);
+};
+
+/* TODO: Maybe we can do this smarter without declaring everything from scratch.
+   We just need to state that the bs.module is `react-navigation` instead of `react-native`
+   */
+module FlatList = {
+  include ReactNative.FlatList;
+
+  [@react.component] [@bs.module "react-navigation"]
```

Comment:
@MoOx Is there a way to achieve this without having to copy-paste the bindings from react-native?

Reply:
It seems you changed to include, which is exactly what you where looking for :)

Reply:
I think I did this previously as well. I essentially need the exact same `makeProps` as the React Native binding, which I guess I can include from ReactNative.FlatList, but I need to override `make` such that it's importing from [@bs.module "react-navigation"] instead of [@bs.module "react-native"]. 

Is there a way I can reference `FlatList.props` that go into `FlatList.make`, such that I could do:

```reason
[@bs.module "react-navigation"]
external make: (FlatList.props) => FlatList.element = "FlatList";
```

or is there a way I could override bs.module?

Reply:
Not sure you can easily do that. We already have to do some copy pasting to achieve the zero-cost in lots of place.
Copy pasting is not a big deal as soon it's documented somewhere. You can probably copy paste & add this kind of comment https://github.com/reasonml-community/reason-react-native/blob/9327dd58a5c56659259f43a78e57f2069b259793/reason-react-native/src/components/FlatList.re#L17

Reply:
Copy paste is ok. To my understanding it seems we're only missing a type for the props we can reference, which react-ppx doesn't appear to give us. I might look at adding this if it's missing.